drivers/flash/CMakeLists.txt

```diff
@@ -22,6 +22,12 @@ zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI flash_mcux_flexspi.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_ESP32 flash_esp32.c)
 
+if(CONFIG_XIP AND CONFIG_FLASH_MCUX_FLEXSPI_NOR AND (CONFIG_CODE_FLEXSPI OR CONFIG_CODE_FLEXSPI2))
```

Comment:
I think it would be better to add a new `CONFIG_FLASH_MCUX_FLEXSPI_XIP` and move this logic to Kconfig defaults. That would also lend well to moving line 28 to the hal module.

Reply:
I will change this and make it depend on `CONFIG_CODE_RELOCATION` as well.

Is it safe to assume `ITCM` memory will always be available?